src/transformStateWithClones.js

```diff
@@ -6,8 +6,44 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
   // write code here
+  const array = [];
+  const stateCopy = { ...state };
+  let pushedCopy = {};
+
+  for (const object of actions) {
+    switch (object.type) {
+      case 'addProperties':
+      // tried spread operator, but somethig is off with it,
+      // so let's go with for loop.
+
+        for (const key in object.extraData) {
+          stateCopy[key] = object.extraData[key];
+        }
```

Comment:
Avoid nasted loops instead, you can use `Object.assign`.

Reply:
@MarcinLigmann We have to make some workarounds in order to use forEach on objects.
Also keep in mind that forEach is also a type of loop ;)